modal/functions.py

```diff
@@ -87,6 +92,27 @@
     import modal.stub
 
 
+class _SynchronizedQueue:
+    """mdmd:hidden"""
+
+    # small wrapper around asyncio.Queue to make it cross-thread compatible through synchronicity
+    async def init(self):
+        # in Python 3.8 the asyncio.Queue is bound to the event loop on creation
+        # so it needs to be created in a synchronicity-wrapped init method
+        self.q = asyncio.Queue()
```

Comment:
In more recent versions you could just create the Queue instance in a constructor (regardless which thread or event loop the constructor ran in) and it will be lazily bound to the first event loop it's used from instead, so a bit cleaner. 
Not 100% sure, but based on some googling it looks like it wasn't actually changed until 3.10, so we would still have to keep it around while we support 3.9